osh/cmd_eval.py

```diff
@@ -208,25 +208,33 @@ def PlusEquals(old_val, val):
                 str_to_append = cast(value.Str, UP_val)
                 val = value.Str(old_val.s + str_to_append.s)
 
-            elif tag == value_e.BashArray:
+            elif tag in [value_e.BashArray, value_e.SparseArray]:
```

Comment:
ditto: please use `()` instead of `[]` , since mycpp has a special case for Python tuples vs. lists